# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/stf/m2vrequantiser/gcc34.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

Nur in M2VRequantiser-patch: a.out.
Nur in M2VRequantiser-patch: M2VRequantiser.
diff -dur M2VRequantiser/main.c M2VRequantiser-patch/main.c
--- M2VRequantiser/main.c	2004-01-21 23:48:32.000000000 +0000
+++ M2VRequantiser-patch/main.c	2004-07-17 20:02:33.000000000 +0000
@@ -12,7 +12,7 @@
 
 #define NDEBUG // turns off asserts
 #define REMOVE_BYTE_STUFFING	// removes series of 0x00
-#define USE_GLOBAL_REGISTER // assign registers to bit buffers
+//#define USE_GLOBAL_REGISTER // assign registers to bit buffers
 #define MAX_ERRORS 0 // if above copy slice
 
 //#define CHANGE_BRIGHTNESS //add a param to command line, changing brightness: _will_not_recompress_, disables max_errors
@@ -120,7 +120,7 @@
 	register int	inbitcnt	asm ("r13");
 	register int	outbitcnt   asm ("r14");
 	register uint32	inbitbuf	asm ("r15");
-	register uint32	outbitbuf   asm ("r16");
+	register uint32	outbitbuf	asm ("r16");
 #else
 	static int		inbitcnt, outbitcnt;
 	static uint32	inbitbuf, outbitbuf;
Nur in M2VRequantiser-patch: main.c~.
Nur in M2VRequantiser-patch: makeTable.
diff -dur M2VRequantiser/makeTable.c M2VRequantiser-patch/makeTable.c
--- M2VRequantiser/makeTable.c	2003-12-30 05:17:01.000000000 +0000
+++ M2VRequantiser-patch/makeTable.c	2004-07-17 20:10:57.000000000 +0000
@@ -18,8 +18,12 @@
 	int alt_table = 0;
 	int fill_table = 1;
 	
-	if (argc != 3) return 2;
-	
+	if (argc != 3) 
+	{	
+		printf("Usage: %s { norm | alt | run } OutputFile\n", argv[0]);
+		return 2;
+	}
+
 	if (strcmp(argv[1], "norm") == 0) alt_table = 0;
 	else if (strcmp(argv[1], "alt") == 0) alt_table = 1;
 	else if (strcmp(argv[1], "run") == 0) fill_table = 0;
Nur in M2VRequantiser-patch: makeTable.c~.
Nur in M2VRequantiser-patch: test.txt.
